storm-webapp/src/main/java/org/apache/storm/daemon/logviewer/webapp/LogviewerResource.java

```diff
@@ -255,8 +302,16 @@ public Response deepSearch(@PathParam("topoId") String topologyId,
         String callback = request.getParameter(StormApiResource.callbackParameterName);
         String origin = request.getHeader("Origin");
 
-        return logSearchHandler.deepSearchLogsForTopology(topologyId, user, searchString, numMatchesStr, portStr,
-                startFileOffset, startByteOffset, BooleanUtils.toBooleanObject(searchArchived), callback, origin);
+        Boolean alsoSearchArchived = BooleanUtils.toBooleanObject(searchArchived);
```

Comment:
I'm sorry I don't understand what you mean by it.

Reply:
I mean the metric update here. If you want to track how many times deepSearchLogsForTopology is called with certain parameters, the metric update should happen in that function. Otherwise, if I add another call to deepSearch elsewhere, I'd have to remember to copy this code too.

Reply:
Though unlikely I see your concerns.

Reply:
I think the intention here is to track number of API invocation, not the actual method invocation.

Reply:
It's probably fine.